### PR TITLE
Tavern GraphQL Rust Client

### LINKED ISSUE #185: Imix implement graphql code generation for callback functions.
**Is your feature request related to a problem? Please describe.**
Currently updating the API requires manual updates to the agent code base.
In order to improve this process we can leverage code generation to create structs defining the graphql schema.

**Describe the solution you'd like**
- Derive graphql objects from the realm common schema.
- GraphQL agents:
  - https://docs.rs/graphql_client/latest/graphql_client/
  - https://docs.rs/cynic/latest/cynic/

**Describe alternatives you've considered**
N/a

**Additional context**
N/a

### PR DESCRIPTION
**Pull request by KCarretto**
Introduces a new `tavern` rust crate that provides a transport agnostic Tavern GraphQL client for agents with a default http implementation. It uses code generation to build new types from our GraphQL schema.

#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

Fixes #185

### CODE REVIEW

**Comment by hulto** [comment]
Very pogggggg taking a look.

File: implants/pkg/tavern/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Tavern (Rust GraphQL Client)
+
+## Overview
+
+* This client library is intended to provide a minimal set of types and operations for agents to interact with the Tavern GraphQL API
+* GraphQL types are generated using the `graphql-client generate` command, provided by the `graphql_client_cli` crate
+  * `codegen.sh` will install and run this command with the correct parameters
+  * The Tavern (server implementation) Go generate will automatically populate the graphql schema and run this command when the schema changes
+* Mutations and the data requested from the server are present in the `graphql/mutations.graphql` file
+* An HTTP transport is included as a default transport (`tavern::http`, feature `http`), however custom implementors of `tavern::Executor` may be defined and used with `tavern::Client`
```
Comment (hulto) [inline implants/pkg/tavern/README.md:10]: Very slick 😎

File: implants/pkg/tavern/graphql/mutations.graphql [hunk 2]
```diff
@@ -0,0 +1,34 @@
+mutation ClaimTasks($input: ClaimTasksInput!) {
```
Comment (hulto) [inline implants/pkg/tavern/graphql/mutations.graphql:1]: Thoughts on reusing the existing schema instead of copying a special one down to imix?

Comment (hulto) [reply implants/pkg/tavern/graphql/mutations.graphql:1]: Ah I see it in go generate

**Review by hulto — APPROVED** [review]
LGTM

File: implants/pkg/tavern/src/http.rs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+
+impl Transport {
+    /*
+     * Prepares a new default HTTP transport using the provided API endpoint.
+     */
+    pub fn new(url: &str, auth_token: &str) -> Result<Self> {
+        let client = reqwest::Client::builder()
+                        .timeout(Duration::from_secs(DEFAULT_TIMEOUT))
+                        .danger_accept_invalid_certs(true)
```
Comment (Cictrone) [inline implants/pkg/tavern/src/http.rs:36]: 😟

**Pull request merged**